sardana_limaccd/ctrl/LimaCCDCtrl.py

```diff
@@ -58,121 +63,144 @@
-        #                 repr(props))
-
+CTRL_PROPERTIES = {
+    'LimaCCDDeviceName': {Type: str, Description: 'Detector device name'},
+    'LatencyTime': {Type: float, Description: 'Maximum latency time'},
+    'FirstImageNumber': {Type: int,
+                         Description: 'First value of the saving next number'}
+}
+
+CTRL_ATTRIBUTES = {
+    'CameraModel': {
+        Type: str,
+        Description: 'LimaCCD attribute camera_model',
+        Access: DataAccess.ReadOnly,
+        Memorize: NotMemorized},
+    'CameraType': {
+        Type: str,
+        Description: 'LimaCCD attribute camera_type',
+        Access: DataAccess.ReadOnly,
+        Memorize: NotMemorized},
+    'SavingCommonHeader': {
+        Type: [str, ],
+        Description: 'LimaCCD attribute saving_common_header',
+        Access: DataAccess.ReadWrite,
+        Memorize: Memorized},
+    'SavingFramePerFile': {
+        Type: int,
+        Description: 'LimaCCD attribute saving_frame_per_file',
+        Access: DataAccess.ReadWrite,
+        Memorize: Memorized,
+        DefaultValue: 1},
+    'SavingHeaderDelimiter': {
+        Type: [str, ],
+        Description: 'LimaCCD attribute saving_header_delimiter',
+        Access: DataAccess.ReadWrite,
+        Memorize: Memorized},
+    'SavingManagedMode': {
+        Type: str,
+        Description: 'LimaCCD attribute saving_managed_mode',
+        Access: DataAccess.ReadWrite,
+        Memorize: Memorized},
+    'SavingMaxWritingTask': {
+        Type: int,
+        Description: 'LimaCCD attribute saving_max_writing_task',
+        Access: DataAccess.ReadWrite,
+        Memorize: Memorized},
+    'SavingMode': {
+        Type: str,
+        Description: 'LimaCCD attribute saving_mode. The controller will'
+                     'set it to AUTO_FRAME by default.',
+        Access: DataAccess.ReadWrite,
+        Memorize: NotMemorized,
+        DefaultValue: 'AUTO_FRAME'},
+    'SavingNextNumber': {
+        Type: int,
+        Description: 'LimaCCD attribute saving_next_number',
+        Access: DataAccess.ReadWrite,
+        Memorize: Memorized},
+    'SavingOverwritePolicy': {
+        Type: str,
+        Description: 'LimaCCD attribute saving_overwrite_policy',
+        Access: DataAccess.ReadWrite,
+        Memorize: Memorized,
+        DefaultValue: 'ABORT'},
+    'SavingFormatsAllowed': {
+        Type:  [str, ],
+        Description: 'Detector SavingModes allowed saving_suffix',
+        Access: DataAccess.ReadOnly,
+        Memorize: NotMemorized},
+    'SavingImageHeaders': {
+        Type: [str, ],
+        Description: 'Headers for each image',
+        Access: DataAccess.ReadWrite,
+        Memorize: NotMemorized,
+        MaxDimSize: (1000000,)},
+    }
+
+
+class LimaImageFormat(struct.Struct):
+
+    Magic = 0x44544159
+    DTypes = ('u1', 'u2', 'u4', None, 'i1', 'i2', 'i4')
+    DTypeSize = (1, 2, 4, None, 1, 2, 4)
+
+    def __init__(self):
+        super().__init__('<IHHIIHHHHHHHHIIIIIIII')
```

Comment:
Should this controllers be python2 compatible? In this case, we need to fix the super calls, adding the class and self instance as arguments.

Reply:
Yes, the idea is to have the same version on py2 and py3 if it does not block the development.

Reply:
Ok I can do that.
I completely forgot this was to be compatible with py2

Reply:
Ok, latest commit should fix plenty of things including this one